pkg/stores/proxy/proxy_store.go

```diff
@@ -505,6 +505,9 @@ func (s *Store) Update(apiOp *types.APIRequest, schema *types.APISchema, params
 		return nil, nil, fmt.Errorf("metadata.resourceVersion is required for update")
 	}
 
+	gvk := attributes.GVK(schema)
+	input["apiVersion"], input["kind"] = gvk.ToAPIVersionAndKind()
```

Comment:
What if both or one of these fields are specified in the body?  Maybe it should be something like:
```
   apiVersion, kind := gvk.ToAPIVersionAndKind()
   v, ok := input["apiVersion"]
   if !ok || v == "" { input["apiVersion"] = apiVersion }
   v, ok = input["kind"]
   if !ok || v == "" { input["kind"] = kind }
```